packages/xl-ai/src/api/formats/html-blocks/defaultHTMLPromptBuilder.ts

```diff
@@ -0,0 +1,111 @@
+      role: "system",
+      content: `You're manipulating a selected part of a text document using HTML blocks. 
+      Make sure to follow the json schema provided and always include the trailing $ in ids. 
+      List items are 1 block with 1 list item each, so block content \`<ul><li>item1</li></ul>\` is valid, but \`<ul><li>item1</li><li>item2</li></ul>\` is invalid. We'll merge them automatically.
+      This is the selection as an array of html blocks:`,
+    },
+    {
+      role: "system",
+      content: JSON.stringify(opts.htmlSelectedBlocks),
+    },
+
+    {
+      role: "system",
+      content:
+        "This is the entire document (INCLUDING the selected text), find the selected text in there to understand the context:",
+    },
+    {
+      role: "system",
+      content: JSON.stringify(opts.htmlDocument),
+    },
+    {
+      role: "system",
+      content: "The user asks you to do the following:",
+    },
+    {
+      role: "user",
+      content: opts.userPrompt,
+    },
+  ];
+}
+
+function promptManipulateDocumentUseHTMLBlocks(opts: {
+  userPrompt: string;
+  htmlBlocks: Array<
+    | {
+        id: string;
+        block: string;
+      }
+    | {
+        cursor: true;
+      }
+  >;
+}): Array<CoreMessage> {
+  return [
+    {
+      role: "system",
+      content: `You're manipulating a text document using HTML blocks. 
+        Make sure to follow the json schema provided. When referencing ids they MUST be EXACTLY the same (including the trailing $). 
+        List items are 1 block with 1 list item each, so block content \`<ul><li>item1</li></ul>\` is valid, but \`<ul><li>item1</li><li>item2</li></ul>\` is invalid. We'll merge them automatically.
+        For code blocks, you can use the \`data-language\` attribute on a code block to specify the language.
+        This is the document as an array of html blocks (the cursor is BETWEEN two blocks as indicated by cursor: true):`,
+    },
+    {
+      role: "system",
+      content: JSON.stringify(opts.htmlBlocks),
+    },
+    {
+      role: "system",
+      content: "The user asks you to do the following:",
+    },
+    {
+      role: "user",
+      content: opts.userPrompt,
+    },
+    {
+      role: "system",
+      content: `First, determine what part of the document the user is talking about. You SHOULD probably take cursor info into account if needed.
+       EXAMPLE: if user says "below" (without pointing to a specific part of the document) he / she probably indicates the block(s) after the cursor. 
+       EXAMPLE: If you want to insert content AT the cursor position (UNLESS indicated otherwise by the user), then you need \`referenceId\` to point to the block before the cursor with position \`after\` (or block below and \`before\`).
+      
+      Prefer updating existing blocks over removing and adding (but this also depends on the user's question).`,
+    },
+  ];
+}
+
+export const defaultHTMLPromptBuilder: PromptBuilder = async (editor, opts) => {
```

Comment:
@nperez0111 as discussed on slack:

> I think it would be even nicer if you can provide a function that takes data similar to promptManipulateSelectionHTMLBlocks, but then we’d need to call getPromptData automatically and also make that optionally configurable. Now you just need to call the helpers yourself (similar to how defaultHTMLPromptBuilder does this)
> I think if we generalize this further it might get quite complicated considering the difference between selection / non selection, and different formats